### DISCUSSION TITLE
Connection testing for API initialization and checking for connection drops in queries

### DISCUSSION THREAD

**Discussion by tghpereira** [prisma/prisma#11917] [Q&A] [answered]
I would first like to congratulate the teams involved in this prisma orm project.

I start this topic by aborting the possibility of prisma orm to offer connection tests for high availability apis with multiple databases, currently I work around this problem by validating errors and executing the connection forwarding but it is often a cost for my api, there is a possibility of creating this feature with a prisma middleware?

example:

`prisma.$use(async (connections, failed, params, next) => {
  if(failed.database.name === 'PRIMARY_DATABASE'){
   await connections.$use('SECONDARY_DATABASE', ({params, newConnection}) => ({
    set: newConnection.$retryQuery(params)
  }))
  }
 return next(params)
})`

**Comment by nurul3101** [comment, marked as answer]
Hey @tghpereira 👋🏻 ,
Can you please create a [Feature Request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) describing your detailed use case so that our Product and Engineering teams can evaluate this? 
Also it would be helpful if you could describe why the prisma middlewares doesn't perfectly fit in your scenario.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
